pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -311,6 +323,179 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// Schedules [DelegationAction::Revoke] for the delegator, towards all delegated collator.
+	/// The last fulfilled request causes the delegator to leave the set of delegators.
+	pub(crate) fn delegator_schedule_revoke_all(
+		delegator: T::AccountId,
+	) -> DispatchResultWithPostInfo {
+		let mut state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+		let mut updated_scheduled_requests = vec![];
+		let now = <Round<T>>::get().current;
+		let when = now.saturating_add(T::LeaveDelegatorsDelay::get());
+
+		// pre-validate that not all delegations already have a Revoke request.
+		let mut revoke_count = 0;
+		for bond in &state.delegations.0 {
+			let collator = bond.owner.clone();
+			let scheduled_requests = <DelegationScheduledRequests<T>>::get(&collator);
+			let maybe_revoke = scheduled_requests.iter().find(|req| {
+				req.delegator == delegator && matches!(req.action, DelegationAction::Revoke(_))
+			});
+			if let Some(_) = maybe_revoke {
+				revoke_count += 1;
+			}
+		}
+		if revoke_count == state.delegations.0.len() {
+			return Err(<Error<T>>::DelegatorAlreadyLeaving.into());
+		}
+
+		// it is assumed that a multiple delegations to the same collator does not exist, else this
+		// will cause a bug - the last duplicate delegation update will be the only one applied.
+		for bond in state.delegations.0.clone() {
+			let collator = bond.owner;
+			let mut scheduled_requests = <DelegationScheduledRequests<T>>::get(&collator);
```

Comment:
`cancel_request_with_state` does not modify the test, so we might not need two loops for this